skimage/morphology/rolling_ball.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+
+import numpy as np
+
+from skimage.util import invert
+
+
+def rolling_ball(input_img, radius=50, white_background=False):
+    """Perform background subtraction using the rolling ball method.
+
+    The rolling ball filter is a segmentation method that aims to separate the
+    background from a grayscale image in case of uneven exposure. It is
+    frequently used in biomedical image processing and was first proposed by
+    Stanley R. Sternberg (1983) in the paper Biomedical Image Processing [1]_.
+
+    Parameters
+    ----------
+    image : ndarray
+        The image to be filtered.
+    radius : float
+        The radius of the ball that is used as the structuring element.
+    white_background : bool
+        If true, the algorithm separates dark features from a bright
+        background.
+
+    Returns
+    -------
+    filtered_image : ndarray
+        The image with background removes
+    background : ndarray
+        The background that was removed.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from skimage import data
+    >>> from skimage.morphology import rolling_ball
+
+    Subtract a black background from an example image
+
+    >>> image = data.coins()
+    >>> result, bg = rolling_ball(image, radius=100)
+
+    Subtract a white background from an example image
+
+    >>> image = data.page()
+    >>> result, bg = rolling_ball(image, radius=100, white_background=True)
+
+    Notes
+    -----
+    Check out the gallery example of this function (TODO: link)
+
+    References
+    ----------
+    .. [1] Sternberg, Stanley R. "Biomedical image processing." Computer 1
+           (1983): 22-34. :DOI:`10.1109/MC.1983.1654163`
+    """
+
+    if not isinstance(radius, float):
+        try:
+            radius = float(radius)
+        except ValueError:
+            raise ValueError(f"Radius should be float, was {type(radius)}")
+
+    if radius <= 0:
+        raise ValueError(f"Radius must be greater zeros, was {radius}")
+
+    try:
+        white_background = bool(white_background)
+    except ValueError:
+        raise ValueError(
+            f"white_background must be convertible to a boolean, "
+            f"was {type(white_background)}"
+        )
+
+    if not isinstance(input_img, np.ndarray):
+        raise ValueError(
+            f"input_img must be a np.ndarray, "
+            f"was {type(input_img)}"
+        )
+
+    if not issubclass(input_img.dtype.type, np.integer):
+        raise ValueError("Currently only integer images are supported.")
+
+    img = input_img.copy()
+    if white_background:
+        img = invert(img)
+
+    # sagitta assuming the position is where the ball touches
+    # the image umbra
+    L = np.arange(-radius, radius + 1)
+    X, Y = np.meshgrid(L, L)
+    distance = np.sqrt(X ** 2 + Y ** 2)
+    sagitta = radius - np.sqrt(radius ** 2 - distance ** 2)
+    sagitta[np.isnan(sagitta)] = np.Inf
```

Comment:
I don't really understand why this is necessary...? `distance**2` is guaranteed to be smaller than `radius**2` except in the corners of the grid, where it *might* be bigger by a small amount due to floating point error. In that case, I think the correct thing to do would be

```python
sagitta = radius - np.sqrt(
        np.clip(radius ** 2 - distance ** 2, 0, None)
)
```
?

Reply:
This is a residual of me experimenting with different ways to split the computation of the filter into chunks. 

If we apply the kernel to the sum of the pixel's intensity and the value of the sagitta (`np.min((img + sagitta) * kernel)`) instead of doing `np.min(img * kernel + sagitta)` we can do it in the way you suggest.